### PR TITLE
modal window

### PR DESCRIPTION
**Pull request by kseniyaLine**
create a template for modal windows

### CODE REVIEW

**Comment by mtalstykh-exadel** [comment]
I'm tired too :) Anyway do not forget about naming

File: src/components/modal/Modal.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        </div>
+    );
+};
+
+Modal.propTypes = {
+    active: PropTypes.bool,
+    setActive: PropTypes.func,
+    children: PropTypes.string
+};
+export default Modal;
```
Comment (sadoffToxir) [inline src/components/modal/Modal.js:20]: add a new line at the end of the file

File: src/App.js [hunk 2]
```diff
@@ -1,40 +1,21 @@
...
+        <div>Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</div>
+        <div>Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. </div>
+        <div>Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</div>
+      </main>
+      <Modal active={modalActive} setActive={setModalActive}>
+        <p>Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
+      </Modal>
     </div>
   );
 }
```
Comment (sadoffToxir) [inline src/App.js:40]: add a new line at the end of the file

File: src/components/modal/Modal.css [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  background-color: blueviolet;
+  border-radius: 15px;
+  transform: scale(0.5);
+  transition: 0.4s all;
+  width: 50vw;
+}
+
+.modal__content.active {
+  transform: scale(1);
+}
```
Comment (sadoffToxir) [inline src/components/modal/Modal.css]: use scss intead of css

File: src/App.js [hunk 4]
```diff
@@ -1,40 +1,21 @@
...
-  }, testButtons: {
-        '&': {
-          margin: 40,
-        }
-      },
-  }));
-
-  const classes = useStyles();
-
+  const [modalActive,setModalActive]=useState(true);
```
Comment (fshovchko) [inline src/App.js]: line should look like this: const [modalActive, setModalActive] = useState(true);

File: src/App.js [hunk 5]
```diff
@@ -1,40 +1,21 @@
...
-      {/* Adding the imported element */}
-      <Button variant="contained" color="primary" disableElevation className={classes.testButtons}>
-        Example button Material-UI
-      </Button>
-      <Button variant="contained" disabled className={classes.testButtons}>
-        Example №2 disabled
-      </Button>
-      <SimpleCounter />
+      <main>
+        <button className="open-button" onClick={()=>setModalActive(true)}>Открыть модальное окно</button>
```
Comment (fshovchko) [inline src/App.js]: Line should look like this:  onClick={() => setModalActive(true)}

File: src/components/modal/Modal.css (hunk 3 above)
Comment (fshovchko) [reply src/components/modal/Modal.css]: > use scss intead of css

Modal.css to Modal.scss

**Review by tensenrey — COMMENTED** [review]
don't push system files of your idea

File: .gitignore [hunk 6]
```diff
@@ -21,3 +21,19 @@ package-lock.json
...
+  }
+};
+
+/*eslint-disable*/
+module.exports = {
+  messages: {
+    "Example button Material-UI": "Пример кнопки Material-UI",
+    "Example №2 disabled": "Пример №2 disabled"
+  }
+};
```
Comment (mtalstykh-exadel) [inline .gitignore:39]: looks like a mistake

File: src/components/modal/Modal.scss [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  top: 0;
+  left: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  transform: scale(0);
+  transition: 0.5;
+}
+
+.modal.active {
```
Comment (mtalstykh-exadel) [inline src/components/modal/Modal.scss]: use SCSS features

File: package.json [hunk 8]
```diff
@@ -7,19 +7,15 @@
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^11.2.7",
     "@testing-library/user-event": "^12.8.3",
-    "formik": "^2.2.9",
```
Comment (mtalstykh-exadel) [inline package.json:10]: why do you delete it?)

File: src/App.js [hunk 9]
```diff
@@ -1,25 +1,43 @@
...
-import { Switch, Route } from 'react-router-dom';
+import SimpleCounter from './store/store';
+
 import './App.scss';
-import Buttons from './components/buttons/Buttons';
-import Counter from './components/counter/counter';
-import PrivateRoute from './utils/privateRoute';
-import Profile from './components/profile/profile';
+
+// importing the elements Material-UI
```
Comment (mtalstykh-exadel) [inline src/App.js:6]: remove the comment. It is not useful

File: src/App.js [hunk 10]
```diff
@@ -1,25 +1,43 @@
...
+import Button from '@material-ui/core/Button';
+import { makeStyles } from '@material-ui/core';
+// brief entry: import {makeStyles, Button} from '@material-ui/core/';
```
Comment (mtalstykh-exadel) [inline src/App.js:9]: same here

**Comment by mtalstykh-exadel** [comment]
will use a modal from material

**Pull request closed without merging**